components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+    public void setUp() {
+
+        daoImpl = new ActionManagementDAOImpl();
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        identityUtil = mockStatic(IdentityUtil.class);
+        actionSecretProcessor = mock(ActionSecretProcessor.class);
+        secretManager = mock(SecretManagerImpl.class);
+        ActionMgtServiceComponentHolder.getInstance().setSecretManager(secretManager);
+        try{
+            identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getDBConnection(anyBoolean()))
+                        .thenReturn(getConnection());
+            identityTenantUtil.when(()-> IdentityTenantUtil.getTenantId(anyString())).thenReturn(TENANT_ID);
+            SecretType secretType = mock(SecretType.class);
+            when(secretType.getId()).thenReturn("secretId");
+            when(secretManager.getSecretType(any())).thenReturn(secretType);
+        }
+        catch (Exception ignored){
+        }
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        identityTenantUtil.close();
+        identityDatabaseUtil.close();
+        identityUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+        closeH2Database();
+    }
+
+    @Test
+    public void testAddAction() throws ActionMgtException {
+
+        Action action = getAction("120120");
+        Action result = daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getDescription(),result.getDescription());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testAddActionWithoutOptional() throws ActionMgtException {
+
+        Action action = getActionWithoutOptionals("136136");
+        Action result = daoImpl.addAction(action.getType().getActionType(), action.getId(), action,TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testGetActionsByActionType() throws ActionMgtException{
+
+        Action action = getAction("149149");
+        Assert.assertEquals(5,daoImpl.getActionsByActionType(action.getType().getActionType(),TENANT_ID).size());
```

Comment:
```suggestion
        Assert.assertEquals(5,daoImpl.getActionsByActionType("PRE_ISSUE_ACCESS_TOKEN", TENANT_ID).size());
```

Just pass the required action type (e.g., `PRE_ISSUE_ACCESS_TOKEN`)  to the getActionsByActionType() to fetch the actions.